src/cardano_fmt_plug.c

```diff
@@ -0,0 +1,237 @@
+/*
+ * JtR format to crack password protected Cardano 128-byte length legacy secret Keys (a.k.a XPrv).
+ *
+ * This software is Copyright (c) 2022, Pal Dorogi <pal dot dorogi at gmail.com>
+ * and it is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ *
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_cardano;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_cardano);
+#else
+
+#include <string.h>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#define OMP_SCALE 1
+
+#include "cardano_common.h"
+#include "blake2.h"
+#include "pbkdf2_hmac_sha512.h"
+#include "chacha.h"
+#include "ed25519.h"
+
+#define FORMAT_NAME         "Cardano Encrypted 128-byte Secret Key (a.k.a XPrv)"
+#define FORMAT_LABEL        "cardano"
+#ifdef SIMD_COEF_64
+#define ALGORITHM_NAME      "PBKDF2-SHA512/BLAKE2b/ChaCha20 " SHA512_ALGORITHM_NAME
+#else
+#if ARCH_BITS >= 64
+#define ALGORITHM_NAME      "PBKDF2-SHA512/BLAKE2b/ChaCha20  64/" ARCH_BITS_STR
+#else
+#define ALGORITHM_NAME      "PBKDF2-SHA512/BLAKE2b/ChaCha20  32/" ARCH_BITS_STR
+#endif
+#endif
+#define BENCHMARK_COMMENT   ""
+#define BENCHMARK_LENGTH    0x107
+#define BINARY_SIZE         0
+#define BINARY_ALIGN        1
+#define SALT_SIZE           sizeof(struct custom_salt)
+#define SALT_ALIGN          sizeof(uint32_t)
```

Comment:
`SALT_ALIGN` can now be 1 now that you removed the `uint32_t` field from there.

Reply:
We would need it as I reverted the 1.

Reply:
You didn't reintroduce the struct field, so we can have `SALT_ALIGN` at 1.